From 06d2e58518f668299eccef79fa9b8d03a270e1a1 Mon Sep 17 00:00:00 2001
From: Cedric Bondier <cedricx.bondier@intel.com>
Date: Thu, 20 Dec 2012 11:08:11 +0100
Subject: [PORT FROM MAIN][BT] Fix sign propagation issue due to cast to (int)

BZ: 74811

Due to integer compilation difference between Arm (unsigned)
and x86 (signed), the cast to (int) could induce a sign propagation
that would create a stack corruption.
To fix this issue, we specify precisely the cast to (UINT8),
in order to be sure of the outcome.

Change-Id: I99ed6f132d5c16e2bbacc922f2d43d2ad4f3e8d6
Orig-Change-Id: I2f1e36f672e579df06b0d9c416a3d8a4cedfdc9b
Feature: BT
Category: Integration
Purpose: PV-Enable
Origin: Internal
Upstream-candidate: yes
Signed-off-by: Cedric Bondier <cedricx.bondier@intel.com>
Reviewed-on: http://android.intel.com:8080/82859
Reviewed-by: Bransiec, ChristopheX <christophex.bransiec@intel.com>
Reviewed-by: Aquilone, RaffaeleX <raffaelex.aquilone@intel.com>
Reviewed-by: Biselli, MirkoX <mirkox.biselli@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 btif/src/btif_util.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/btif/src/btif_util.c b/btif/src/btif_util.c
index 636c1bc..49ef385 100755
--- a/btif/src/btif_util.c
+++ b/btif/src/btif_util.c
@@ -96,8 +96,8 @@ char *bd2str(const bt_bdaddr_t *bdaddr, bdstr_t *bdstr)
     char *addr = (char *) bdaddr->address;
 
     sprintf((char*)bdstr, "%02x:%02x:%02x:%02x:%02x:%02x",
-                       (int)addr[0],(int)addr[1],(int)addr[2],
-                       (int)addr[3],(int)addr[4],(int)addr[5]);
+                       (UINT8)addr[0],(UINT8)addr[1],(UINT8)addr[2],
+                       (UINT8)addr[3],(UINT8)addr[4],(UINT8)addr[5]);
     return (char *)bdstr;
 }
 
-- 
1.7.4.1

